runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -80,9 +78,9 @@ class MM_ObjectAllocationAPI
 	initializeContiguousIndexableObject(J9VMThread *currentThread, bool initializeSlots, J9Class *arrayClass, uint32_t size, uintptr_t dataSize, j9object_t *objectHeader)
 	{
```

Comment:
seems like objectHeader could be passed by value, in which case, the code could be simpler:

```
initializeContiguousIndexableObject(J9VMThread *currentThread, bool initializeSlots, J9Class *arrayClass, uint32_t size, uintptr_t dataSize, j9object_t objectHeader)
		bool isCompressedReferences = J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread);
		void *dataAddr = (void *)((uintptr_t)objectHeader + _sizeOfJ9IndexableObjectContiguous);

		if (isCompressedReferences) {
			J9IndexableObjectContiguousCompressed *header = (J9IndexableObjectContiguousCompressed*)objectHeader;
			header->clazz = (uint32_t)(uintptr_t)arrayClass;
			header->size = size;
		} else {
			J9IndexableObjectContiguousFull *header = (J9IndexableObjectContiguousFull*)objectHeader;
			header->clazz = (uintptr_t)arrayClass;
			header->size = size;
		}

#if defined(J9VM_ENV_DATA64)
		if (_isDataAddrPresent) {
			header->dataAddr = dataAddr;
		}
#endif /* J9VM_ENV_DATA64 */
		initializeIndexableSlots(initializeSlots, dataSize, dataAddr);
}
```

Reply:
went too far... setting dataAddr has to be cloned too, but initializeIndexableSlots could still be shared

Reply:
Yes not a problem, I've done that 👍